test/testProxy.js

```diff
@@ -0,0 +1,68 @@
+/*global require, exports*/
+
+const { makePeer, RNode, RHOCore, h2b } = require('..');
+
+const { fromJSData } = RHOCore;
+const def = Object.freeze;
+
+/**
+ * To run this test, first deploy target1.rho (and propose).
+ * It will register a contract and print out the registry URI.
+ *
+ * Pass that URI on the command line.
+ */
+
+
+async function test({ rnode, clock, user, setTimeout, target }) {
+  const peer = makePeer(rnode, user, clock, setTimeout);
```

Comment:
Making a `peer` only to later make separate proxies seems clunky. That may be because I don't really understand what `user` is and why you've chosen to make a `peer` encapsulate it. Why not just pass user in to the RNode or each proxy depending what it represents?

Reply:
Good point.

In fact, one approach is to include `makeProxy` in `RNode` along with `doDeploy` and the rest..